### PR TITLE
Fire UIA event on Checkbox state change

### PR DESCRIPTION
**Pull request by yaakovschectman**
When a checkbox has its state changed, fire a UIA property change event for the toggle state property, so that screen readers can read the updated state.

Part of https://github.com/flutter/flutter/issues/119350. I still need to confirm what the appropriate behavior should be when hitting NVDA+UP when focusing on a checkbox before I can potentially close that issue.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by cbracken — APPROVED** [review]
Except for the usual bot complaints about formatting; looks great.

File: third_party/accessibility/ax/platform/ax_platform_node_win.cc [hunk 1]
```diff
@@ -5266,7 +5266,6 @@ std::optional<EVENTID> AXPlatformNodeWin::MojoEventToUIAEvent(
   }
 }
 
-// static
```
Comment (yaakovschectman) [inline third_party/accessibility/ax/platform/ax_platform_node_win.cc:5269]: Need to make this a method as the property may depend on the role (e.g. for `kValueChanged`)

**Comment by yaakovschectman** [comment]
From the issue:
```
- When the user pressed NVDA+UP with a checkbox focused, the checkbox info is not read aloud.
  - This should be the case, as NVDA+UP is only meant to read textual information. NVDA+TAB is a separate keyboard command for reading the current focus, and it does in fact read the checkbox state.
```
This means the issue can be resolved if this PR lands.

**Review by loic-sharma — APPROVED** [review]
Nice work! 🎉

**Pull request merged**